src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -196,6 +198,80 @@ public void GenerateHeader()
             PopBlock();
         }
 
+
+        private class DelegateSignature
+        {
+            public string Signature { get; set; }
+            public bool SuppressWarning { get; set; }
+            public string CallingConvention { get; set; }
+            public string StrReturnType { get; set; }
+        }
+
+        private class OptionalOut<Type>
+        {
+            public List<Type> Result { get; set; }
+        }
+        private DelegateSignature GenerateDelegateSignature(Function function, bool suppressWarning, out CSharpTypePrinterResult retType, out string delegateName)
+        {
+            OptionalOut<string> allTypes = new OptionalOut<string>();
+            allTypes.Result = new List<string>();
+            var @params = GatherInternalParams(function, out retType, false, allTypes);
+
+            DelegateSignature ds = new DelegateSignature();
+            ds.CallingConvention = function.CallingConvention.ToInteropCallConv().ToString();
+            ds.SuppressWarning = suppressWarning;
+
+            allTypes.Result.Add(ds.CallingConvention);
+            allTypes.Result.Add(suppressWarning.ToString());
```

Comment:
We don't need the value of "suppressWarning" in the delegate signature. It only indicates whether to put the SuppressWarning attribute which is ignored when invalid unlike the calling convention which causes a crash.